File: le-spark/src/test/resources/scripts/calc_pi.scala
val NUM_SAMPLES = lattice.params.get("NUM_SAMPLES").asInt()
val count = sc.parallelize(1 to NUM_SAMPLES).filter { _ =>
  val x = math.random
  val y = math.random
  x*x + y*y < 1
}.count()

lattice.outputStr = s"Pi is roughly ${4.0 * count / NUM_SAMPLES}"

File: le-spark/src/test/resources/scripts/calc_pi.py
import random

NUM_SAMPLES = lattice.params['NUM_SAMPLES']

def inside(p):
    x, y = random.random(), random.random()
    return x*x + y*y < 1

# -----CELL BREAKER----

count = sc.parallelize(range(0, NUM_SAMPLES)).filter(inside).count()

lattice.output_str = "Pi is roughly %f" % (4.0 * count / NUM_SAMPLES)
